Evolutionary_loop/Loop_Parts/Part_D/part_d_job1.py

```diff
@@ -0,0 +1,114 @@
+#*******************************************************************************
+# File: part_d_job1.py
+#    Based on Part_D1_Array.sh.
+#
+#  Programmer: Jason Yao (yao.966@osu.edu)
+#
+#  Revision history:
+#     07/28/23 Jason Yao, Original version
+#
+#  Notes:
+#       * vertical ruler at column 80
+#  TODO:
+#       * Incorporate process 0 into part_c.py
+#       * p1 needs to be combined with the sbatch command (ie. in the same
+#         subprocess)
+#       * remove .root files
+#
+#*******************************************************************************
+'''
+This is Part D1 of the Loop.
+This script moves each .dat file (from Part C) into a folder AraSim can access,
+while changing it to a .txt, which is what AraSim reads. For each individual,
+the script then runs AraSim and moves the output into Antenna_Performance_Metric.
+Finally, this script makes a directory for all errors and output files from
+AraSim to be dumped.
+
+Usage: python4 part_d_job1.py <indiv> <gen> <npop> <working_dir> ... <nsections>
+
+For more information on the arguments: 
+  >> python3 part_d_job1.py -h
+'''
+
+
+import argparse
+import subprocess as sp
+
+from pathlib import Path
+
+
+def main(gen, npop, working_dir, arasim_exec,
+         exp, nnt, run_name, seeds, debug_mode):
+    
+## AraSim SETUP
+    specific_seed = 32000
+
+    metric_path = working_dir / 'Antenna_Performance_Metric'
+    for i in range(1,npop+1):
+        p0 = sp.run(f'mv {metric_path}/evol_antenna_model_{i}.dat '
+                    f'{arasim_exec}/a_{i}.txt', shell=True, capture_output=True)
+        if p0.stderr:
+            print('Error at start of part_d_job1.py when moving .dat files')
+    
+    # p1 = sp.run(f'source {arasim_exec}/new_root_setup.sh\n'
+    #             f'source cvmfs/ara.opensciencegrid.org/rrunk/centos7/setup.sh',
+    #             shell=True)
+    
+    if not debug_mode:
+        # replace num_nnu (number of neutrinots thrown) in setup_dummy.txt
+        p2 = sp.run(f'sed -e "s/num_nnu/{nnt}/" -e "s/n_exp/{exp}/" '
+                    f'-e "s/current_seed/{specific_seed}/" '
+                    f'{arasim_exec}/setup_dummy_araseed.txt '
+                    f'>> {arasim_exec}/setup.txt',
+                    shell=True, capture_output=True, text=True)
```

Comment:
This should also follow the approach discussed in #6 if possible.

Reply:
[c48ea9d](https://github.com/osu-particle-astrophysics/GENETIS_HPol/pull/8/commits/c48ea9dc90363d5ea24430819dde6e417db35a4e)